src/Domain/Operation/isOperationPermitted.ts

```diff
@@ -0,0 +1,22 @@
+import { UserPermissions } from '../User/UserWithPermissions'
+import { Operation } from './Operation'
+import { Resource } from '../Resource/Resource'
+
+// todo: implement based on what auth is capable of returning; + tests
+/**
+ * Determines if the given permissions allow executing the requested operation.
```

Comment:
Don't leave comments in code. Code should be self-explanatory.

Reply:
😮 Is this a blanket philosophy @karolsojko? What about comments for functions? I like when I hover over a function from a caller interface for example and see what the function does without having to read its code.

Reply:
@mobitar as for "blanket philosophy":

“The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures.”
― Robert C. Martin, Clean Code: A Handbook of Agile Software Craftsmanship

Reply:
Sure, I will be lighter on the comments and actively remove the ones people think don't add value (this one is indeed unnecessary). But a general zero comment policy I think is harmful.

I think there is plenty of space for sanity in between hardcore self-documenting code and hardcore literate programming.

What Martin says here is hyperbole which he immediately tones down. He goes on to write a whole chapter about comments, starting with the "Good Comments" section. A direct quote:

"Some comments are necessary or beneficial."

So clearly, even according to him comments are not always bad.

Another popular author on the topic, Steve McConnell even prescribes "integrating commenting into your development style". He does however acknowledge that good comments are difficult to write and commenting can be damaging. And I agree.

All this is to say that I agree with a no-unnecessary-comment* policy, but I think a zero-comment policy is too extreme.

* heck, I agree with a no-unnecessary-anything policy -- I think we should have as little lines of code (and comments, and other writing) as sensible -- achieving that though is an art in itself.